src/services/homework/hooks/index.js

```diff
@@ -94,127 +110,208 @@ const hasViewPermissionAfter = (hook) => {
-					)
-				));
-				const submissionWithGrade = filteredSubmission.filter((submission) => submission.grade);
-				if (submissionWithGrade.length === 1) {
-					copy.grade = submissionWithGrade[0].grade.toFixed(2);
-				}
+				if (!isTeacher(hook.params.account.userId, copy)) {
+					const currentSubmissions = submissions.data.filter((s) =>
+						equalIds(copy._id, s.homeworkId),
+					);
+					const filteredSubmission = currentSubmissions.filter(
+						(submission) =>
+							equalIds(
+								hook.params.account.userId,
+								submission.studentId,
+							) ||
+							(submission.teamMembers &&
+								submission.teamMembers.some((member) =>
+									equalIds(
+										hook.params.account.userId,
+										member,
+									),
+								)),
+					);
+					const submissionWithGrade = filteredSubmission.filter(
+						(submission) => submission.grade,
+					);
+					if (submissionWithGrade.length === 1) {
+						copy.grade = submissionWithGrade[0].grade.toFixed(2);
+					}
 
-				copy.hasEvaluation = false;
-				if (filteredSubmission.filter(isGraded).length === 1) {
-					copy.hasEvaluation = true;
-				}
+					copy.hasEvaluation = false;
+					if (filteredSubmission.filter(isGraded).length === 1) {
+						copy.hasEvaluation = true;
+					}
 
-				copy.submissions = (filteredSubmission.filter(isValidSubmission)).length;
-			}
+					copy.submissions = filteredSubmission.filter(
+						isValidSubmission,
+					).length;
+				}
 
-			if (!copy.private && (
-				(((copy.courseId || {}).userIds || []).includes(hook.params.account.userId.toString()) && copy.publicSubmissions)
-                || isTeacher(hook.params.account.userId, copy))) {
-				const NumberOfCourseMembers = ((copy.courseId || {}).userIds || []).length;
-				const currentSubmissions = submissions.data.filter((s) => equalIds(copy._id, s.homeworkId));
-				const validSubmissions = currentSubmissions.filter(isValidSubmission);
-				const gradedSubmissions = currentSubmissions.filter(isGraded);
-				const NumberOfUsersWithSubmission = validSubmissions.map((e) => (e.courseGroupId ? ((e.courseGroupId.userIds || []).length || 1) : ((e.teamMembers || []).length || 1))).reduce((a, b) => a + b, 0);
+				if (
+					!copy.private &&
+					((((copy.courseId || {}).userIds || []).includes(
+						hook.params.account.userId.toString(),
+					) &&
+						copy.publicSubmissions) ||
+						isTeacher(hook.params.account.userId, copy))
+				) {
+					const NumberOfCourseMembers = (
+						(copy.courseId || {}).userIds || []
+					).length;
+					const currentSubmissions = submissions.data.filter((s) =>
+						equalIds(copy._id, s.homeworkId),
+					);
+					const validSubmissions = currentSubmissions.filter(
+						isValidSubmission,
+					);
+					const gradedSubmissions = currentSubmissions.filter(
+						isGraded,
+					);
+					const NumberOfUsersWithSubmission = validSubmissions
+						.map((e) =>
```

Comment:
Codacy found an issue: ['e' is already declared in the upper scope.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5987920)